packages/@aws-cdk/cdk/lib/aspects/aspect.ts

```diff
@@ -0,0 +1,27 @@
+import { IConstruct } from '../core/construct';
+
+export enum AspectVisitType {
+  /**
+   * The aspect will visit each Construct only once
+   */
+  Single = 'Single',
+
+  /**
+   * The aspect will visit each contruct as manay times as invoked
+   */
+  Multiple = 'Multiple'
+}
+
+/**
+ * Represents an Aspect
+ */
+export interface IAspect {
+  /**
+   * Aspects invocation pattern
+   */
+  readonly visitType: AspectVisitType;
```

Comment:
First, with the move of aspect visit only once to the framework (`ConstructNode`) the need for a base `Aspect` is gone. Speak up if you disagree.

With all the simplification (thanks team for the comments), I no longer need a string type here. @eladb and I were talking about multiple visit pattern and I added it here. Looking at `prepareTree()` and where this sits I don't think the multiples make a lot of sense.  Two questions for your feedback:
1. Should I bring back the string type?
1. Should we keep `AspectVisitType`?

Reply:
It's a little confusing to me. What's the point of the multiple visit type? When will it be useful?

Reply:
ready to remove -- see comments on the README section.